src/shared/apparmor-util.c

```diff
@@ -7,6 +7,39 @@
 #include "fileio.h"
 #include "parse-util.h"
 
+#if HAVE_APPARMOR
+static void *libapparmor_dl = NULL;
+
+DLSYM_PROTOTYPE(aa_change_onexec) = NULL;
+DLSYM_PROTOTYPE(aa_change_profile) = NULL;
+DLSYM_PROTOTYPE(aa_features_new_from_kernel) = NULL;
+DLSYM_PROTOTYPE(aa_features_unref) = NULL;
+DLSYM_PROTOTYPE(aa_policy_cache_dir_path_preview) = NULL;
+DLSYM_PROTOTYPE(aa_policy_cache_new) = NULL;
+DLSYM_PROTOTYPE(aa_policy_cache_replace_all) = NULL;
+DLSYM_PROTOTYPE(aa_policy_cache_unref) = NULL;
+
+int dlopen_libapparmor(void) {
+        ELF_NOTE_DLOPEN("apparmor",
+                        "Support for AppArmor policies",
+                        ELF_NOTE_DLOPEN_PRIORITY_RECOMMENDED,
```

Comment:
why recommended and not suggested?

i get the sense that systemd should never pull in libapparmor with default pkg manager settings. I mean, this stuff won't do anyone good unless they actually install policies. And that step of installing policies should pull in the libapparmor, but not systemd, since all alone that's just gonna be dead weight.

hence suggests sounds more appropriate to me?

Reply:
Hmm, apart from pkg managers this could also be parsed by initrd generators. My thinking has been that this is core functionality (initialized by pid1 at very early stage) and almost everyone has libapparmor installed would have a policy (Arch OTOH _never_ automatically install optional deps, so it's not a problem there). But I'm not religious about it. Happy to change if people disagree.

Also, I was trying to follow @keszybz's #32971 here.